testutils/src/main/java/com/microsoft/identity/internal/testutils/labutils/ConfidentialClientHelper.java

```diff
@@ -60,6 +68,35 @@ abstract TokenRequest createTokenRequest()
             throws CertificateException, UnrecoverableKeyException, NoSuchAlgorithmException,
             KeyStoreException, NoSuchProviderException, IOException;
 
+    protected synchronized static String getBuildConfigSecretFromClasspath() {
+        if (sClasspathSecret != null) {
+            return sClasspathSecret;
+        }
+        final Reflections r = new Reflections("com.microsoft");
+        final Set<Class<?>> allClasses = r.getSubTypesOf(Object.class);
+        for (final Class<?> clazz : allClasses) {
+            if (clazz.getSimpleName() == "BuildConfig") {
+                try {
+                    final Field labField = clazz.getDeclaredField("LAB_CLIENT_SECRET");
+                    if (labField != null) {
+                        String secret = (String) labField.get(null);
+                        if (!TextUtils.isEmpty(secret)) {
+                            Logger.info("LabAuthenticationHelper:init", "Found lab secret on class with name " + clazz.getCanonicalName());
+                            sClasspathSecret = secret;
+                            return secret;
+                        }
+                    }
+                    //Swallow all exceptions.
+                } catch (final NoSuchFieldException e) {
+                } catch (final IllegalAccessException e) {
+                } catch (final ClassCastException e) {
```

Comment:
Can we collapse these into one catch block?

Reply:
Yes, just not the way I have my studio set up.  I'll fix it.